rest/importtest/import_test.go

```diff
@@ -1916,6 +1918,191 @@ func TestImportRevisionCopyUnavailable(t *testing.T) {
+	response := rt.SendAdminRequest("GET", "/db/_raw/"+key, "")
+	rest.RequireStatus(t, response, http.StatusNotFound)
+	assertDocProperty(t, response, "reason", "Not imported")
+
+	// assert stat still no incremented as import filter rejects the doc
+	computeStat1 := rt.GetDatabase().DbStats.DatabaseStats.ImportProcessCompute.Value()
+	require.Equal(t, int64(0), computeStat1)
+
+	// rewrite through SG - should treat as new insert
+	docBodyString := `{"type":"SG client rewrite",
+	                 "channels": "NBC"}`
+	response = rt.SendAdminRequest("PUT", fmt.Sprintf("/db/%s", key), docBodyString)
+	rest.RequireStatus(t, response, http.StatusCreated)
+
+	// assert stat increases as function OnDemandImportForWrite will have been executed
+	computeStat2 := rt.GetDatabase().DbStats.DatabaseStats.ImportProcessCompute.Value()
+	require.Greater(t, computeStat2, computeStat1)
+}
+
+func TestAutoImportComputeStat(t *testing.T) {
+	base.SkipImportTestsIfNotEnabled(t)
+
+	rtConfig := rest.RestTesterConfig{
+		SyncFn: channels.DocChannelsSyncFunction,
+		DatabaseConfig: &rest.DatabaseConfig{DbConfig: rest.DbConfig{
+			AutoImport: true,
+		}},
+	}
+	rt := rest.NewRestTester(t, &rtConfig)
+	defer rt.Close()
+	dataStore := rt.GetSingleDataStore()
+
+	key := t.Name()
+	docBody := make(map[string]interface{})
+	docBody["test"] = t.Name()
+	docBody["channels"] = "ABC"
+
+	// assert the stat starts at 0 for a new database
+	computeStat := rt.GetDatabase().DbStats.DatabaseStats.ImportProcessCompute.Value()
+	require.Equal(t, int64(0), computeStat)
+
+	// add doc to bucket
+	_, err := dataStore.Add(key, 0, docBody)
+	require.NoError(t, err, fmt.Sprintf("Unable to insert doc %s", key))
+
+	// wait for import to process
+	_, ok := base.WaitForStat(func() int64 {
+		return rt.GetDatabase().DbStats.SharedBucketImportStats.ImportCount.Value()
+	}, 1)
+	require.True(t, ok)
+
+	// assert the stat increments for the auto import of the above doc
+	computeStat1 := rt.GetDatabase().DbStats.DatabaseStats.ImportProcessCompute.Value()
+	require.Greater(t, computeStat1, int64(0))
+}
+
+func TestQueryResyncImportComputeStat(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("Test requires Couchbase Server")
+	}
+
+	rtConfig := rest.RestTesterConfig{
+		SyncFn: channels.DocChannelsSyncFunction,
+		DatabaseConfig: &rest.DatabaseConfig{DbConfig: rest.DbConfig{
+			AutoImport: false,
+			Unsupported: &db.UnsupportedOptions{
+				UseQueryBasedResyncManager: true,
+			},
+		}},
+	}
+	rt := rest.NewRestTester(t, &rtConfig)
+	defer rt.Close()
+	const numDocs = 1000
+	var resp *rest.TestResponse
+
+	for i := 0; i < numDocs; i++ {
+		resp = rt.SendAdminRequest(http.MethodPut, "/{{.keyspace}}/"+fmt.Sprint(i), `{"random": "doc"}`)
+		rest.RequireStatus(t, resp, http.StatusCreated)
+	}
+
+	computeStat := rt.GetDatabase().DbStats.DatabaseStats.ImportProcessCompute.Value()
+	require.Equal(t, int64(0), computeStat)
+
+	resp = rt.SendAdminRequest(http.MethodPost, "/{{.db}}/_offline", "")
```

Comment:
offline is async so we need to wait for this to actually turn offline, this is copy pasted from elsewhere in the code base but you need:

```
        response = rt.SendAdminRequest("POST", "/db/_offline", "")
	RequireStatus(t, response, 200)

	rt.waitForState("online")
```

I might find the other places we do and replace this with a function like rt.takeDbOffline